README.md

```diff
@@ -16,3 +52,161 @@ pip install cookiecutter
 ```sh
 cookiecutter gh:elixir-cloud-aai/cookiecutter-python
 ```
+
+> **Note**: If you're using just `cookiecutter`, manually remove
+  `update-template` job from `update.yaml` github action workflow file as
+  `cookiecutter` doesn't support autosync like cruft.
+
+## Configurations
+
+### Package manager - Poetry
+
+This project uses [Poetry][poetry-org] as a package manager. Check out the
+commands at the [official documentation][poetry-docs].
+
+### Linters and formatters - Ruff
+
+To lint and format `Python` code files, it uses [Ruff][ruff], the default
+configuration is set in the `pyproject.toml` file.
+
+```toml
+select = [
+  "B", # flake8-bugbear
+  "E", # pycodestyle
+  "F", # Pyflakes
+  "I", # isort
+  "PL", # pylint
+  "SIM", # flake8-simplify
+  "UP", # pyupgrade
+]
+```
+
+To configure it to your needs, refer to the [rules documentation][rudff-linter]
+, and for formatter configuration, refer to the
+[configuration documentation][ruff-formatter].
+
+### Spell checker - Typos
+
+If you want to ignore certain words, add them to the `pyproject.toml` file,
+under the `tool.typos.default.extend-words` key.
+
+```toml
+[tool.typos.default.extend-words]
+mke = 'mke'
+```
+
+For further configuration, refer to the [typos docs][typos-docs].
+
+### Static type checker - Mypy
+
+Change the configuration in `pyproject.toml` file, for further information refer to
+the [documentation][mypy-config].
+
+### Documentation - Sphinx
+
+Configuration for Sphinx is in the `docs/source/conf.py` file, for further info refer
+to the [Sphinx documentation][sphinx-docs]. The default configuration uses
+[furo theme][furo] and [ReadTheDocs][rtd] to host the documentation, you can
+change the configuration using `/docs/source/conf.py` and `.readthedocs.yml`
+file.
+
+> **Note**: Make sure to setup the `ReadTheDocs` account and add the project to
+  the account to host the documentation.
+
+### Testing - pytest and pytest-cov
+
+Generate a coverage report using `pytest-cov` and uploads it to codecov.io.
+
+> **Note**: Make sure to setup the `Codecov` account and add the project to the
+  account to host the coverage report and add github [token](#cicd---github-actions).
+
+## CI/CD - GitHub Actions
+
+Here are the GitHub Actions Secrets that need to be included in the repository
+settings:
+
+- `PYPI_PASSWORD`: PyPI account password.
```

Comment:
I think it would be more accurate to call this `PYPI_TOKEN`. I have now added both `PYPI_TOKEN` and `PYPI_PASSWORD` (for backward compatibility) as organization secrets.

I think it would also be good to add to the description:
> PyPI account token scoped to this repository (requires the project to exist prior to using the Cookiecutter) or to all repositories.

Reply:
Also might be good to include instructions:

> To set up a token, visit https://pypi.org, go to "Account settings" and hit the "Add API token" button in the "API tokens" section. Add a name, choose the scope and create the token.

Reply:
Might be good to include instructions for Codecov as well:

> To set up the token, visit https://codecov.io/, log in, find the repository that you want to set up (you may need to "Resync") and hit the "Configure" button. The token should show up in the "Coverage" tab.